File: lsh/src/services/api/file.ts
import fprint from 'fprint';
import fs from 'fs';

const IN_MEMORY_FILE_FINGERPRINTS = {};
const NO_CHANGE_TO_FILE = -1;

const encodeContent = async (path) => {
  const fileContents = fs.readFileSync(path);
  const fingerprint = await fprint(fileContents, 'md5');

  if (IN_MEMORY_FILE_FINGERPRINTS[path] !== fingerprint) {
    IN_MEMORY_FILE_FINGERPRINTS[path] = fingerprint;
    return fs.readFileSync(path, {
      encoding: 'base64',
    });
  } else {
    return NO_CHANGE_TO_FILE;
  }
};

export const FILE = {
  encodeContent,
  IN_MEMORY_FILE_FINGERPRINTS,
  NO_CHANGE_TO_FILE,
};

File: lpy/src/lpy/c3/file.py
import hashlib
import base64
import os

IN_MEMORY_FILE_FINGERPRINTS = {}
NO_CHANGE_TO_FILE = -1

def encode_content(path):
    print("encode_content")
    print(path)
    with open(path, 'rb') as file:
        print(file)
        # content = file.read()
    # fingerprint = hashlib.md5(content).hexdigest()

    return NO_CHANGE_TO_FILE

    if IN_MEMORY_FILE_FINGERPRINTS.get(path) != fingerprint:
        IN_MEMORY_FILE_FINGERPRINTS[path] = fingerprint
        return base64.b64encode(content).decode('utf-8')
    else:
        return NO_CHANGE_TO_FILE
